### PR TITLE
MAID-1846: Renamed final instances of "group" to "section"

### PR DESCRIPTION
**Pull request by Fraser999**
This breaks the API which Vault uses, but a corresponding PR will be submitted to handle this Vault shortly.

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/lib.rs [hunk 1]
```diff
@@ -24,17 +24,17 @@
...
 //!
-//! Group authorities are also addressed using a single `XorName`. The members of that group are
+//! Section authorities are also addressed using a single `XorName`. The members of that section are
 //! the nodes that are closest to that name. Since nodes are assigned their name by the network,
-//! this provides redundancy and resilience: A node has no control over which group authorities it
-//! will be a member of, and without a majority in the group it cannot forge a message from that
-//! group.
+//! this provides redundancy and resilience: a node has no control over which section authority it
+//! will be a member of, and without a majority in the section it cannot forge a message from that
+//! section.
```
Comment (afck) [inline src/lib.rs]: In general documentation we should probably refer to both groups and sections? Or mention below that the same also applies to groups.

File: src/routing_table/mod.rs [hunk 2]
```diff
@@ -36,8 +36,8 @@
 // The *[XOR][3] distance* between two nodes with addresses `x` and `y` is `x ^ y`. This
 // [distance function][4] has the property that no two points ever have the same distance from a
 // given point, i. e. if `x ^ y == x ^ z`, then `y == z`. This property allows us to define the
-// `k`-*close group* of an address as the `k` closest nodes to that address, guaranteeing that the
-// close group will always have exactly `k` members (unless, of course, the whole network has less
+// `k`-*close section* of an address as the `k` closest nodes to that address, guaranteeing that the
+// close section will always have exactly `k` members (unless, of course, the whole network has less
```
Comment (afck) [inline src/routing_table/mod.rs]: These should be `group`, I think. (Also below.)

File: tests/mock_crust/churn.rs [hunk 3]
```diff
@@ -55,32 +55,32 @@ fn random_churn<R: Rng>(rng: &mut R,
 /// The entries of a Get request: the data ID, message ID, source and destination authority.
 type GetKey = (DataIdentifier, MessageId, Authority<XorName>, Authority<XorName>);
 
-/// A set of expectations: Which nodes and groups are supposed to receive Get requests.
+/// A set of expectations: Which nodes and sections are supposed to receive Get requests.
```
Comment (afck) [inline tests/mock_crust/churn.rs]: Those can be both groups and sections.

File: src/lib.rs (hunk 1 above)
Comment (Fraser999) [inline src/lib.rs]: Yes, I'll rephrase this part.

File: src/states/node.rs [hunk 4]
```diff
@@ -61,7 +61,7 @@ use xor_name::XorName;
 const TICK_TIMEOUT_SECS: u64 = 60;
 /// Time (in seconds) after which a `GetNodeName` request is resent.
 const GET_NODE_NAME_TIMEOUT_SECS: u64 = 60;
-/// Time (in seconds) the new close group waits for a joining node it sent a network name to.
+/// Time (in seconds) the new close section waits for a joining node it sent a network name to.
```
Comment (fizyk20) [inline src/states/node.rs]: Like before - "close section" might not be the best phrase.

File: src/states/node.rs [hunk 5]
```diff
@@ -1422,24 +1421,23 @@ impl Node {
                                     message_id: MessageId)
                                     -> Evented<Result<(), RoutingError>> {
         let hashed_key = sha256::hash(&client_key.0);
-        let close_group_to_client = XorName(hashed_key.0);
+        let close_section_to_client = XorName(hashed_key.0);
 
-        // Validate Client (relocating node) has contacted the correct Group-X
-        if close_group_to_client != dst_name {
+        // Validate Client (relocating node) has contacted the correct Section-X
+        if close_section_to_client != dst_name {
```
Comment (fizyk20) [inline src/states/node.rs]: Again - maybe "matching section"?

File: tests/ci.rs [hunk 6]
```diff
@@ -341,11 +344,11 @@ fn core() {
...
 
     {
-        // request to group authority
+        // request to section authority
         let node_names = nodes.iter().map(|node| node.name()).collect_vec();
-        let client = TestClient::new(nodes.len(), event_sender.clone(), min_group_size);
+        let client = TestClient::new(nodes.len(), event_sender.clone(), min_section_size);
         let data = gen_structured_data(client.full_id(), &mut rng);
-        let mut close_group = closest_nodes(&node_names, client.name(), min_group_size);
+        let mut close_section = closest_nodes(&node_names, client.name(), min_section_size);
```
Comment (fizyk20) [inline tests/ci.rs]: I think this is actually still a close group.

File: tests/ci.rs [hunk 7]
```diff
@@ -374,15 +377,15 @@ fn core() {
...
 
     {
-        // response from group authority
+        // response from section authority
         let node_names = nodes.iter().map(|node| node.name()).collect_vec();
-        let client = TestClient::new(nodes.len(), event_sender.clone(), min_group_size);
+        let client = TestClient::new(nodes.len(), event_sender.clone(), min_section_size);
         let data = gen_structured_data(client.full_id(), &mut rng);
-        let mut close_group = closest_nodes(&node_names, client.name(), min_group_size);
+        let mut close_section = closest_nodes(&node_names, client.name(), min_section_size);
```
Comment (fizyk20) [inline tests/ci.rs]: Like above - still a group.

File: tests/mock_crust/churn.rs [hunk 8]
```diff
@@ -230,28 +231,28 @@ fn churn() {
...
-        expected_gets.send_and_expect(data_id, auth_n0, auth_n1, &mut nodes, min_group_size);
-        expected_gets.send_and_expect(data_id, auth_n0, auth_g0, &mut nodes, min_group_size);
-        expected_gets.send_and_expect(data_id, auth_n0, auth_s0, &mut nodes, min_group_size);
-        // ... and from a group to itself, another group, a section and a node...
-        expected_gets.send_and_expect(data_id, auth_g0, auth_g0, &mut nodes, min_group_size);
-        expected_gets.send_and_expect(data_id, auth_g0, auth_g1, &mut nodes, min_group_size);
-        expected_gets.send_and_expect(data_id, auth_g0, auth_s0, &mut nodes, min_group_size);
-        expected_gets.send_and_expect(data_id, auth_g0, auth_n0, &mut nodes, min_group_size);
-        // ... and from a section to itself, another section, a group and a node...
+        // Test messages from a node to itself, another node, a section and a section...
```
Comment (fizyk20) [inline tests/mock_crust/churn.rs]: This should be still a "group" ("a section and a section" has a nice ring to it, though :grin:). Similarly below.

File: tests/mock_crust/drop.rs [hunk 9]
```diff
@@ -20,11 +20,11 @@ use routing::{Event, EventStream};
 use routing::mock_crust::{Config, Endpoint, Network};
 use super::{TestNode, create_connected_nodes, poll_all, verify_invariant_for_all_nodes};
 
-// Drop node at index and verify its close group receives NodeLost.
+// Drop node at index and verify its close section receives NodeLost.
```
Comment (fizyk20) [inline tests/mock_crust/drop.rs]: Again: "close section" doesn't sound good to me.

File: tests/mock_crust/mod.rs [hunk 10]
```diff
@@ -185,29 +185,29 @@ fn simultaneous_joining_nodes() {
 }
 
 #[test]
-fn check_close_groups_for_group_size_nodes() {
-    let min_group_size = 8;
-    let nodes = create_connected_nodes(&Network::new(min_group_size, None), min_group_size);
-    let close_groups_complete = nodes.iter()
-        .all(|n| nodes.iter().all(|m| m.close_group().contains(&n.name())));
-    assert!(close_groups_complete);
+fn check_close_sections_for_section_size_nodes() {
```
Comment (fizyk20) [inline tests/mock_crust/mod.rs]: Again "close sections".

File: tests/mock_crust/utils.rs [hunk 11]
```diff
@@ -145,7 +145,7 @@ impl TestNode {
         unwrap!(self.inner.name())
     }
 
-    pub fn close_group(&self) -> HashSet<XorName> {
+    pub fn close_section(&self) -> HashSet<XorName> {
```
Comment (fizyk20) [inline tests/mock_crust/utils.rs]: Again "close section".

File: tests/mock_crust/churn.rs (hunk 8 above)
Comment (Fraser999) [inline tests/mock_crust/churn.rs]: Oops! :D

**Pull request merged**